core/src/main/java/org/eclipse/hono/util/ResourceLimits.java

```diff
@@ -41,21 +42,46 @@
      * 
      * @param maxConnections The maximum number of connections to set.
      * @return A reference to this for fluent use.
+     * @throws IllegalArgumentException if the maximum connections is set to less than -1.
      */
     public ResourceLimits setMaxConnections(final int maxConnections) {
+        if (maxConnections < -1) {
+            throw new IllegalArgumentException("maximum Connections can be set to >= -1");
+        }
         this.maxConnections = maxConnections;
         return this;
     }
 
     /**
      * Gets the maximum number of connected devices a tenant supports.
      * 
-     * @return The maximum number of connections.
+     * @return The maximum number of connections or {@link TenantConstants#DEFAULT_MAX_CONNECTIONS}
+     *         if not set.
      */
     public int getMaxConnections() {
         return this.maxConnections;
     }
 
+    /**
+     * Gets the data volume properties which are required for the message limit verification.
+     *
+     * @return The data volume properties.
+     */
+    public DataVolume getDataVolume() {
```

Comment:
FMPOV all of the getters and setter should be declared final as there is no obvious reason for subclasses to override them, is there?

Reply:
`Tenant` and classes used within `Tenant` don't follow this pattern and hence I got bit influenced from it. I don't think there is any obvious reason to override. I will declare the getters and setters as _final_ in `ResourceLimits`, `DataVolume` and `DataVolumePeriod`.